### PR TITLE
Add window_url attribute to Map

### PR DESCRIPTION
**Pull request by davidbrochart**
It can be useful to retrieve the URL of the browser window when we want to serve local tiles with `LocalTileLayer` through a custom Jupyter server handler.

### CODE REVIEW

**Comment by davidbrochart** [comment]
Ideally, I would like the window URL to be retrieved only once, but I don't quite know where it should be. Right now it will be retrieved each time the map bounds are changed.

File: js/src/Map.js [hunk 1]
```diff
@@ -107,6 +108,7 @@ export class LeafletMapModel extends widgets.DOMWidgetModel {
       this.set('south', bounds.south);
       this.set('east', bounds.east);
       this.set('west', bounds.west);
+      this.set('window_url', window.location.href);
```
Comment (martinRenou) [inline js/src/Map.js:111]: You can put this line where the map is initialized?

Comment (davidbrochart) [reply js/src/Map.js:111]: You mean in the `initialize` method of the view?

Comment (martinRenou) [reply js/src/Map.js:111]: It would be even better in the initialize method of the model, then it is called only once.

Comment (davidbrochart) [reply js/src/Map.js]: Thanks.

File: js/src/Map.js [hunk 2]
```diff
@@ -34,7 +34,6 @@ export class LeafletMapModel extends widgets.DOMWidgetModel {
       _model_module: 'jupyter-leaflet',
       _view_module: 'jupyter-leaflet',
 
-      window_url: '',
```
Comment (martinRenou) [inline js/src/Map.js]: Shouldn't we keep this?

Comment (davidbrochart) [reply js/src/Map.js]: It doesn't seem to be needed.

Comment (martinRenou) [reply js/src/Map.js]: It is needed when having an embedded widget, as it retrieves the model from here. Anyway as we discussed this PR won't make sense in the case of an embedded widget.

**Pull request merged**